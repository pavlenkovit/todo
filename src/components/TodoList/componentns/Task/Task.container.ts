import { connect } from 'react-redux';

import Task from './Task';
import { deleteTask, toggleTask } from '../../../../store/actions/tasks';

export interface IDispatchFromPropsTask {
  onToggle: (id: number) => void;
  onDelete: (id: number) => void;
}

const mapDispatchToProps = (dispatch: any) => {
  return {
    onToggle: (id: number) => dispatch(toggleTask({ id })),
    onDelete: (id: number) => dispatch(deleteTask({ id })),
  };
};

export default connect<null, IDispatchFromPropsTask, void>(
  null, mapDispatchToProps
)(Task);
