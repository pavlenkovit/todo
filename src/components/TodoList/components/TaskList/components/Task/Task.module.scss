.item {
  display: flex;
  align-items: center;
  padding: 10px 15px;
  margin: 0 -15px;
  cursor: pointer;
  font-size: 20px;
  border-radius: 2px;
  &.completed {
    text-decoration: line-through;
    & .checkbox {
      & svg {
        opacity: 1;
      }
    }
  }
  &:hover {
    background: #cbcbcb;
  }
}

.checkbox {
  border: solid 2px #000;
  border-radius: 2px;
  padding: 3px 2px 2px;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 24px;
  width: 24px;
  min-width: 24px;
  margin-right: 10px;
  box-sizing: border-box;
  & svg {
    opacity: 0;
    width: 100%;
  }
}

.title {
  flex: 1 1 auto;
}

.delete {
  width: 24px;
  min-width: 24px;
  height: 24px;
  border: none;
  background: none;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-left: 10px;
  padding: 0;
  cursor: pointer;
  opacity: 0.35;
  & svg {
    width: 14px;
  }
  &:hover {
    opacity: 1;
  }
}
