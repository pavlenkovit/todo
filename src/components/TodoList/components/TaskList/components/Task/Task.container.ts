import { connect } from 'react-redux';

import Task from './Task';
import { deleteTask, toggleTask } from '../../../../../../store/actions/tasks';

export interface IDispatchFromPropsTask {
  toggleTask: (id: number) => void;
  deleteTask: (id: number) => void;
}

const mapDispatchToProps = { toggleTask, deleteTask };

export default connect<null, IDispatchFromPropsTask, void>(
  null, mapDispatchToProps,
)(Task);
