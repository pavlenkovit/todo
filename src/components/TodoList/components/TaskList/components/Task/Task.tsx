import React from 'react';
import cn from 'classnames';

import styles from './Task.module.scss';
import { ITask } from '../../../../../../models/tasks';
import { IDispatchFromPropsTask } from './Task.container';
import CheckMark from '../../../../../../icons/CheckMark';
import Close from '../../../../../../icons/Close';

export interface ITaskProps extends ITask, IDispatchFromPropsTask {
}

const Task: React.SFC<ITaskProps> = ({ id, title, completed, onToggle, onDelete }) => {
  const handleClickOnItem = () => {
    onToggle(id);
  };

  const handleClickOnDeleteButton = () => {
    onDelete(id);
  };

  return (
    <li className={cn(styles.item, { [styles.completed]: completed })} onClick={handleClickOnItem}>
      <div className={styles.checkbox}>
        <CheckMark />
      </div>
      <div className={styles.title}>{title}</div>
      <button className={styles.delete} onClick={handleClickOnDeleteButton}>
        <Close />
      </button>
    </li>
  );
};

export default Task;
