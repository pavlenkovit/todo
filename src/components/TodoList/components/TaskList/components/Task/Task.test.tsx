import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Task, { ITaskProps } from './Task';

configure({ adapter: new Adapter() });

describe('Task', () => {
  const props: ITaskProps = {
    userId: 1,
    id: 101,
    title: 'explicabo enim',
    completed: false,
    onDelete: () => {},
    onToggle: () => {},
  };
  const wrapper = shallow(<Task {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('title', () => {
    expect(
      wrapper
        .find('.title')
        .first()
        .text()
    ).toEqual(props.title);
  });
});
