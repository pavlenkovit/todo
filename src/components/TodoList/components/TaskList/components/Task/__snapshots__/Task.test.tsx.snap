// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders properly 1`] = `ShallowWrapper {}`;
